components/style/properties/longhand/font.mako.rs

```diff
@@ -344,6 +344,73 @@ ${helpers.single_keyword("font-variant",
     }
 </%helpers:longhand>
 
+<%helpers:longhand products="gecko" name="font-synthesis" animatable="False">
+    use cssparser::ToCss;
+    use std::fmt;
+    use values::LocalToCss;
+    use values::computed::ComputedValueAsSpecified;
+    use values::NoViewportPercentage;
+
+    impl ComputedValueAsSpecified for SpecifiedValue {}
+    impl NoViewportPercentage for SpecifiedValue {}
+
+    pub mod computed_value {
+        pub use super::SpecifiedValue as T;
+    }
+
+    #[derive(Debug, Clone, PartialEq)]
+    #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+    pub struct SpecifiedValue {
+        weight: bool,
+        style: bool,
+    }
+
+    impl ToCss for computed_value::T {
+        fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+            if self.weight {
```

Comment:
This function generates incorrect output when both `self.weight` and `self.style` are true (it'll only generate `"weight"`).